pkg/user/user.go

```diff
@@ -69,3 +69,11 @@ type Manager interface {
 	GetUserGroups(ctx context.Context, uid *userpb.UserId) ([]string, error)
 	FindUsers(ctx context.Context, query string) ([]*userpb.User, error)
 }
+
```

Comment:
I've used an extra interface to facilitate the rpc communication. The only diff is that this interface has an extra `New` method to initialize the plugins

Reply:
Uhm, this in non ideal. I would expect concrete types to use New to create an object that implement the interface.

With this approach,  the concrete object (after has been created) has a New method, this is very confusing.

Reply:
@labkode I know this is a bit weird and confusing, but since we're using RPCs for communication, there is no way to initialize the object that implements the interface i.e no way to configure the plugin. The `New` method is to pass the configuration (the `user.demo.json` file in case of json plugin.) Without this method, we cannot pass the config, rendering the subsequent methods useless.

Reply:
To avoid confusion, we can maybe change the function name to something more appropriate like `Initialize()`?

Reply:
@labkode, this has been updated!